src/utils/network-monitor.ts

```diff
@@ -911,7 +925,12 @@ export class NetworkMonitor {
         this.structuredLog(job.network, `Block processing complete ✅`, job.block)
       }
 
-      if (this.parent.id === 'indexer' || this.parent.id === 'operator') {
+      if (
+        (this.parent.id === 'indexer' || this.parent.id === 'operator') &&
+        this.apiService !== undefined &&
+        this.blockHeightOptions !== undefined &&
+        this.blockHeightOptions === BlockHeightOptions.API
```

Comment:
I think part of this has to be `isSaveBlockHeightEnabled` right?

Reply:
No, but I just add one more function for it.